src/client.cpp

```diff
@@ -113,18 +113,29 @@ Client::Client(
 	m_script->setEnv(&m_env);
 }
 
-void Client::loadMods()
+void Client::loadBuiltin()
 {
 	// Load builtin
 	scanModIntoMemory(BUILTIN_MOD_NAME, getBuiltinLuaPath());
 
-	// If modding is not enabled, don't load mods, just builtin
+	m_script->loadModFromMemory(BUILTIN_MOD_NAME);
```

Comment:
If builtin mod is loaded even if other mods are not, builtin mod needs to be checked for integrity (that means, there should be a hash that builtin is checked against; and the compiling pipeline needs to be changed to compute this hash and define it for C++ to know) -- the point being that it should not be possible to modify the builtin mod code without recompiling.

Reply:
@Ekdohibs i approve

Reply:
Yes please, some way to avoid abuse of builtin seems essential.

Reply:
I futher suggest not checking the builtin integrity unless absolutely necessary (requested by server), so people can modify builtin in other cases.

Reply:
@sfan5 can you precise the conditions you think about ?

Reply:
what Imean is: <b>only</b> check builtin integrity if the server indicates `CSM_FL_LOAD_CLIENT_MODS`

Reply:
then if server blocks CSM mods loading, verify the integrity, we agree ?

Reply:
Yes that sounds fine